### PR TITLE
Bug 1819461: validate Azure regions

### PR DESCRIPTION
**Pull request by fabianofranz**
Regions were already being filtered by two rules: the regions available for the subscription in context,
and the ones that support the creation of resources by the given subscription. That filtering however
happened only at the survey CLI level, meaning invalid regions would not be displayed as an option, but would
be accepted if the users provided, for example, their own install-config file specifying an invalid region,
resulting in later stages of the resources creation.

This change applies the same filtering to actual validations and adds proper error messages in case invalid
regions are provided.

### CODE REVIEW

**Comment by fabianofranz** [comment]
@jhixson74 @abhinavdahiya for review and approval, thanks!

File: pkg/asset/installconfig/azure/validation.go [hunk 1]
```diff
@@ -69,3 +72,44 @@ func validateMachineNetworksContainIP(fldPath *field.Path, networks []types.Mach
...
+
+	availableRegions := map[string]string{}
+	for _, location := range *locations {
+		availableRegions[to.String(location.Name)] = to.String(location.DisplayName)
+	}
+
+	if _, ok := availableRegions[p.Region]; !ok {
+		errMsg := fmt.Sprintf("region %q is not valid or not available for this account", p.Region)
+
+		lowercaseRegion := strings.ToLower(p.Region)
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/azure/validation.go:91]: https://github.com/terraform-providers/terraform-provider-azurerm/blob/3e8852744df3d18f4d064cc9adf39ea131e96504/azurerm/internal/location/normalize.go#L5-L19

File: pkg/asset/installconfig/azure/validation_test.go [hunk 2]
```diff
@@ -39,7 +45,9 @@ var (
 	invalidateVirtualNetwork     = func(ic *types.InstallConfig) { ic.Azure.VirtualNetwork = "invalid-virtual-network" }
 	invalidateComputeSubnet      = func(ic *types.InstallConfig) { ic.Azure.ComputeSubnet = "invalid-compute-subnet" }
 	invalidateControlPlaneSubnet = func(ic *types.InstallConfig) { ic.Azure.ControlPlaneSubnet = "invalid-controlplane-subnet" }
-	invalidateRegion             = func(ic *types.InstallConfig) { ic.Azure.Region = "eastus" }
+	invalidateRegion             = func(ic *types.InstallConfig) { ic.Azure.Region = "neverland" }
+	invalidateRegionCapabilities = func(ic *types.InstallConfig) { ic.Azure.Region = "australiacentral2" }
+	invalidateRegionLetterCase   = func(ic *types.InstallConfig) { ic.Azure.Region = "CentralUS" }
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/azure/validation_test.go:50]:
```
az account list-locations -otable | grep australiacentral2
Australia Central 2   -35.3075    149.1244     australiacentral2
```
I think the user might give us a display name, which has spaces, so maybe we should test that too.

File: pkg/asset/installconfig/azure/validation.go (hunk 1 above)
Comment (fabianofranz) [reply pkg/asset/installconfig/azure/validation.go]: That's an internal package so it can't be imported, but the logic is simple so I reproduced it here.

**Comment by fabianofranz** [comment]
@abhinavdahiya thanks for the review, comments addressed. PTAL

**Pull request merged**